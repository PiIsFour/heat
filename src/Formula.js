import React, {Component} from 'react'
import PropTypes from 'prop-types'
import './Formula.css'

import loadScript from 'load-script'

const SCRIPT = 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-MML-AM_HTMLorMML'

class Formula extends Component {
	constructor (props) {
		super(props)
		this.state = {
			loaded: false
		}
	}

	componentDidMount () {
		// this.preview.innerHTML = this.props.math
		if (this.state.loaded) {
			window.MathJax.Hub.Queue(['Typeset', window.MathJax.Hub])
		} else {
			loadScript(SCRIPT, (err) => {
				if (err) {
					console.log(err)
				} else {
					this.setState({
						loaded: true
					})
					window.MathJax.Hub.Config({
						showMathMenu: true,
						tex2jax: { inlineMath: [['$', '$'], ['\\(', '\\)']] }
					})
					window.MathJax.Hub.Queue(['Typeset', window.MathJax.Hub])
				}
			})
		}
	}

	componentDidUpdate () {
		window.MathJax.Hub.Queue(['Typeset', window.MathJax.Hub])
	}

	render () {
		if (!this.props.children) {
			return ''
		}
		return <span><math xmlns="http://www.w3.org/1998/Math/MathML">{this.props.children}</math></span>
	}
}

Formula.propTypes = {
	children: PropTypes.node
}

export default Formula
