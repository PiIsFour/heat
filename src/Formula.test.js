import React from 'react'
import renderer from 'react-test-renderer'

import Formula from './Formula'

describe('Formula', () => {
	it('renders and matches our snapshot', () => {
		const component = renderer.create(
			<Formula><mn>1</mn></Formula>
		)
		const tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
