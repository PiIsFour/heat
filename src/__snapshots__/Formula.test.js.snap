// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Formula renders and matches our snapshot 1`] = `
<span>
  <math
    xmlns="http://www.w3.org/1998/Math/MathML"
  >
    <mn>
      1
    </mn>
  </math>
</span>
`;
